<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="UTF-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
      
        <!-- Bootstrap 3 CSS -->
        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
      
        <!-- Custom CSS -->
        <link rel="stylesheet" href="styling/main.css"> 
      
        <title>Flowcharts</title>
      
        <!-- Mermaid -->
        <script type="module">
          import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@10/dist/mermaid.esm.min.mjs';
          mermaid.initialize({ startOnLoad: true });
        </script>
      
        <!-- jquery -->
        <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
      
        <!-- Bootstrap JS -->
        <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/js/bootstrap.min.js"></script>
      </head>    

<body>

<!-- navbar -->
<nav class="navbar navbar-inverse" id="navbar">
    <div class="container-fluid">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#myNavbar">
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="navbar-brand" href="#">Pipeline Flowcharts</a>
        </div>
        <div class="collapse navbar-collapse" id="myNavbar">
            <ul class="nav navbar-nav">
                <li class="dropdown">
                    <a class="dropdown-toggle" data-toggle="dropdown" href="#">MixedPreprocessing <span class="caret"></span></a>
                    <ul class="dropdown-menu">
                        <li><a href="info/inputs.html">Find Files</a></li>
                        <li><a href="info/checkProject.html">Check Project</a></li>
                        <li><a href="info/compile.html">Compile Workflow</a></li>
                        <li><a href="info/update.html">Update Sample Details</a></li>
                        <li><a href="info/setup.html">Setup Batch</a></li>
                        <li><a href="info/run.html">Run Workflow</a></li>
                        <li><a href="info/multiqc.html">MultiQC Report</a></li>
                    </ul>
                </li>
                <li class="dropdown">
                    <a class="dropdown-toggle" data-toggle="dropdown" href="#">EG <span class="caret"></span></a>
                    <ul class="dropdown-menu">
                        <li><a href="info/egStep1.html">EG Step 1</a></li>
                        <li><a href="info/egStep2.html">EG Step 2</a></li>
                    </ul>
                </li>
                <li class="dropdown">
                    <a class="dropdown-toggle" data-toggle="dropdown" href="#">tREX <span class="caret"></span></a>
                    <ul class="dropdown-menu">
                        <li><a href="info/trexStep1.html">tREX Step 1</a></li>
                        <li><a href="info/trexStep2.html">tREX Step 2</a></li>
                    </ul>
                </li>
            </ul>
        </div>
    </div>
</nav>

<div class="main-content">
<!-- Main Content -->
<h1>Interactive Flowcharts for EGL Pipelines</h1>
<h2>Pipelines: MixedPreprocessing, EG, tREX </h2>

<li>Red nodes: tasks in the pipeline. You can click on them for more information.</li>
<li>Blue nodes: outputs of the preceding tasks.</li>

<p>I wrote this site to provide a resource for new bioinformatics trainees who want to better understand how pipelines operate at Exeter Genomics Laboratory. It provides written explanations, making it easier to follow compared to navigating through extensive code. </p>

<p>When using this site for learning, I recommend referencing the corresponding workflow in GitHub at the same time. </p>

<p>GitHub navigation tip: if you look through the main workflow scripts on GitHub, you'll notice that each pipeline starts with an import section. This section helps you quickly navigate to specific pages in the repo: simply copy the relative path from the import section and paste it into your browser as part of the full GitHub URL. </p>

<h2>Mixed Preprocessing</h2>
<p>This pipeline processes all samples together: genomes, exomes, research and targeted NGS samples. <br> Samples are processed on 25B Illumina flowcells on the Novaseq X Plus. This preprocessing pipeline takes the raw BCL outputs and converts them to FASTQs, ready for downstream processing by the EG and tREX pipelines. MultiQC reports are also produced. </p>
<div class="mermaid">
    graph TD
    Inputs["Inputs: project ID and directory, docker image<br>Optional inputs: mode, branch"]:::linked-->findFiles["Find files from runproject<br>(ID & directory)"]:::linked-->fastqArrays["Arrays of: <br>WGS FASTQs & FastQCs <br>trEXome FASTQs & FastQCs"]-->checkProject["Check project"]:::linked-->type["If sample is whole genome, select genome MultiQC<br> If tREX/Exome, select exome MultiQC"]

    click Inputs "../info/inputs.html"
    click findFiles "../info/findFiles.html"
    click fastqArrays "../info/fastqArrays.html"
    click checkProject "../info/checkProject.html"
    
    checkProject["Check project"]:::linked-->compile["Compile workflow in DNAnexus"]:::linked-->
    Update["Update sample details"]:::linked-->Setup["Set up batch<br>"]:::linked-->
    if[if exome or genome, scatter families<br>if tREX, no scattering applied]-->
    run["Run workflow"]:::linked-->
    multiqcf["collect multiqcZipFinal"]-->report["Generate MultiQC report + run metrics"]:::linked

    click compile "../info/compileWorkflow.html"
    click Update "../info/updateDetails.html"
    click Setup "../info/setupBatch.html"
    click run "../info/runWf.html"

    type["If sample is whole genome, select genome MultiQC<br> If tREX/Exome, select exome MultiQC"]-->
    multiQC:::linked-->
    report["Generate MultiQC report + run metrics"]:::linked
    
    click multiQC "../info/multiQC.html"
    click report "../info/finalMultiQC.html"

    classDef default fill:#FFF,stroke:#4499bb,stroke-width:2px,color:#000;
    classDef linked fill:#FFF,stroke:#cc6666,stroke-width:2px;

</div> <br>

<h2>Targeted R-code Exomes (tREX)</h2>
<p>This is the targeted pipeline, which processes exome sequencing data. It uses a virtual capture kit to restrict analysis to the specific regions of the exome that contain the genes included in the specified R-code panel. </p>


    <div class="mermaid">


    </div>

</body>
</div>

<!-- JS for collapsible navbar -->
<script>
    document.addEventListener("DOMContentLoaded", () => {
        const collapsibleButtons = document.querySelectorAll(".collapsible-btn");

        collapsibleButtons.forEach(button => {
            button.addEventListener("click", () => {
                const content = button.nextElementSibling;
                content.style.display = content.style.display === "block" ? "none" : "block";
            });
        });
    });
</script>

</html>
