<!DOCTYPE html>
<html lang="en">
<head>
    <!-- CSS -->
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/css/bootstrap.min.css">
    <link rel="stylesheet" href="../../styling/main.css">
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Find Files</title>
</head>
<div class="main-content">
<body>
    <h1>Find Files</h1>
    <p>The script defines a function called <code>find_files</code>, which searches for files on DNAnexus, extracts file paths, and saves them to output text files. The script then searches for specific files and pulls the following:</p>
    <ul>
        <li>Config files for MultiQC (customize appearance/structure of MultiQC reports)</li>
        <li>Run info files:
            <ul>
                <li><code>runInfo.txt</code>, <code>runParameters.txt</code> (contain metadata about sequencing run)</li>
            </ul>
        </li>
        <li>Run metrics files (for assessing quality/performance):
            <ul>
                <li><code>InteropTar.txt</code> (Interoperability metrics for QC. Includes quality metrics (Q30), intensity data, index read performance, tile metrics, etc.)</li>
                <li><code>metricsXlsx.txt</code>:
                    <ul>
                        <li>Summary: high-level statistics on yield/qual scores/run success</li>
                        <li>Indexing: demultiplexing success rate, index hopping rate</li>
                        <li>Quality metrics: % Q30, per-cycle error rate</li>
                        <li>Flowcell metrics: cluster density, pass-filter %, per-error rates</li>
                    </ul>
                </li>
            </ul>
        </li>
        <li>FASTQ files:
            <ul>
                <li>DRAGEN or non-DRAGEN metrics:
                    <p>DRAGEN is typically used with our NovaSeqs in EGL. DRAGEN is a proprietary software developed by Illumina for use with newer Illumina sequencers, which automatically produces FASTQC files straight from the sequencer. However, occasionally we may use the university's sequencers (e.g., if we're loading many batches and run out of space on our sequencers), and in these cases, we need to use FastQC. More information on this can be found on the <a href="https://github.com/rdemolgen/run_folder_upload">run_folder_upload</a> repo, which is a script for monitoring run folders, running BCL convert (BCL -> FASTQ), and streaming up FASTQs and associated metrics produced by DRAGEN from 25B flowcells. (This of course runs prior to the MixedPreprocessing workflow).</p>
                </li>
                <li>BCL convert metrics:
                    <ul>
                        <li><code>Demultiplex_Stats.txt</code></li>
                        <li><code>Quality_Metrics.csv</code></li>
                        <li><code>Adapter_Metrics.csv</code></li>
                        <li><code>Top_Unknown_Barcodes.csv</code></li>
                    </ul>
                </li>
            </ul>
        </li>
    </ul>

    <h3>BCL Convert</h3>
    <p>BCL convert converts files from their raw binary call format (the initial format generated by an Illumina sequencer) into FASTQs. This process involves demultiplexing (figuring out which reads belong to which sample based on their barcodes). Furthermore, Illumina sequencers score the confidence with which they made a certain call as a Phred-scaled quality score. The CSVs produced by BCL convert provide info on various quality metrics, and these will be incorporated into the FastQC and MultiQC reports.</p>

    <h2>Outputs</h2>
    <ul>
        <li>Configuration files for MultiQC</li>
        <li>Metadata files</li>
        <li>FASTQs and FASTQCs, which are saved as either exome or WGS FASTQs/FASTQCs. Note that the ones labeled 'exome' are for both exomes and targeted samples.</li>
        <li>BCL convert metrics</li>
        <li>Demultiplex stats</li>
    </ul>


    <a href="../../index.html">Back to Flowchart</a>
</body>
</html>